proposals/NNNN-result-codable-and-async-init.md

```diff
@@ -0,0 +1,83 @@
+# Result: Codable conformance & async init
+
+* Proposal: SE-NNNN
+* Author: [Konrad 'ktoso' Malawski](https://github.com/ktoso)
+* Review Manager: TBD
+* Status: **Pending implementation**
+* Implementation: TBD
+
+## Introduction
+
+The `Result` type in Swift is often used to bridge between async and not async contexts, and could use some minor convenience improvements for both `async` contexts as well as encoding/decoding.
+
+## Motivation
+
+The Result type is often used to bridge between async and not async contexts, and it may be useful to initialize a `Result` from the result of an async computation, to then pass it along to other non-async code without unwrapping it first. This is currently inconvenitnt because the `init(catching:)` initializer is missing an `async` overload.
```

Comment:
```suggestion
The Result type is often used to bridge between async and not async contexts, and it may be useful to initialize a `Result` from the result of an async computation, to then pass it along to other non-async code without unwrapping it first. This is currently inconvenient because the `init(catching:)` initializer is missing an `async` overload.
```